src/cmds/pbs_rsub.c

```diff
@@ -696,8 +765,158 @@ main(int argc, char *argv[], char *envp[])
 		}
 	}
 
+	if (is_maintenance_resv) {
+		int i;
+
+		pbs_errno = 0;
+		bstat_head = pbs_stathost(connect, "", NULL, NULL);
+		if (bstat_head == NULL) {
+			if (pbs_errno) {
+				errmsg = pbs_geterrmsg(connect);
+				if (errmsg != NULL) {
+					fprintf(stderr, "pbs_rsub: %s\n", errmsg);
+				} else {
+					fprintf(stderr, "pbs_rsub: Error (%d) submitting reservation\n", pbs_errno);
+				}
+			} else {
+				fprintf(stderr, "pbs_rsub: No nodes found\n");
+			}
+
+			CS_close_app();
+			exit(pbs_errno);
+		}
+
+		/* here, the select is crafted from hosts */
+		hostp = maintenance_hosts;
+		for (;*hostp ;hostp++) {
+			/* lets craft select_str */
+			for (bstat = bstat_head; bstat; bstat = bstat->next) {
+				for (pattr = bstat->attribs; pattr; pattr = pattr->next) {
+				    if (pattr->resource && strcmp(pattr->name, "resources_available") == 0 && strcmp(pattr->resource, "ncpus") == 0)
```

Comment:
Let us use "ATTR_rescavail" macro instead of the string literal.

Reply:
Done